session.cpp

```diff
@@ -139,11 +148,11 @@ shared_ptr<Device> Session::probe_device(libusb_device* device) {
 	}
 
 	if (desc.idVendor == 0x59e3 && desc.idProduct == 0xCEE1) {
-		dev = shared_ptr<Device>(new CEE_Device(this, device));
+		dev = (Device*)(new CEE_Device(this, device));
 	} else if (desc.idVendor == 0x0456 && desc.idProduct == 0xCEE2) {
-		dev = shared_ptr<Device>(new M1000_Device(this, device));
+		dev = (Device*)(new M1000_Device(this, device));
 	} else if (desc.idVendor == 0x064B && desc.idProduct == 0x784C) {
-		dev = shared_ptr<Device>(new M1000_Device(this, device));
+		dev = (Device*)(new M1000_Device(this, device));
```

Comment:
The memory is freed when the last shared_ptr copied from the initial one is deleted.
http://www.cplusplus.com/reference/memory/shared_ptr/

Memory allocated with `new` and not stored in a shared pointer is freed when you explicitly `delete` it.